### PR TITLE
fix(CLOUDDST-24922): retry cosign in signing task

### PR DESCRIPTION
**Pull request by emilyzheng**
Retry failed cosign so that signing doesn't fail with temporary issues on pushing to quay.io.

### CODE REVIEW

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml [hunk 1]
```diff
@@ -24,6 +24,10 @@ spec:
         The relative path in the workspace to a text file that contains a list of repositories
         that needs registry.access.redhat.com image references to be signed (i.e.
         requires_terms=true), one repository string per line, e.g. "rhtas/cosign-rhel9".
+    - name: retries
+      description: Retry cosign N times.
+      type: string
+      default: "0"
```
Comment (scoheb) [inline tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml]: Maybe a default of 3 more practical?

**Comment by johnbieren** [comment]
I'm not sure we should do this before https://issues.redhat.com/browse/RELEASE-1251 is complete

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml (hunk 1 above)
Comment (emilyzheng) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml]: Yeah, changed it to 3.

**Comment by emilyzheng** [comment]
> I'm not sure we should do this before https://issues.redhat.com/browse/RELEASE-1251 is complete

I think [RELEASE-1251](https://issues.redhat.com//browse/RELEASE-1251) is to make it safe to retry the whole rh-sign-image-cosign task (maybe by checking if signatures exist before signing?), which is not conflict with retrying a single failed cosign command here.

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml (hunk 1 above)
Comment (mmalina) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml]: Yeah, I agree. That way you won't have to change the pipeline definition.

**Review by mmalina — APPROVED** [review]

**Review by johnbieren — APPROVED** [review]

**Comment by johnbieren** [comment]
> > I'm not sure we should do this before https://issues.redhat.com/browse/RELEASE-1251 is complete
> 
> I think [RELEASE-1251](https://issues.redhat.com//browse/RELEASE-1251) is to make it safe to retry the whole rh-sign-image-cosign task (maybe by checking if signatures exist before signing?), which is not conflict with retrying a single failed cosign command here.

You're right, sorry for the noise. I saw `retries: 3` in a yaml file and assumed it was a pipeline yaml having the task retry 3x. Approved

**Comment by johnbieren** [comment]
/retest
Failed because multiplatform controller didn't start up

**Pull request merged**